docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +162,240 @@ Note that:
+```
+
+  1) At some point C8Y cloud sends a ConfigUploadRequest for type `bar.conf` for `child1` arrives at C8Y config plugin.
+    The format of C8Y SmartREST message for config retrieval operation is: `526,<childid>,<config type>`. See [C8Y SmartREST doc](https://cumulocity.com/guides/reference/smartrest-two/#upload-configuration-file-with-type-526)<br/>
+    Example: `526,child1,bar.conf`
+
+  2) C8Y config plugin: notifies the external device `child1` via MQTT to upload it's current configuration to the thin-edge device
+
+    Topic: `tedge/configuration/req/retrieve/{config type}/{childid}`<br/>
+    Example: `tedge/configuration/req/retrieve/bar.conf/child1`<br/>  
+    TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307) 
+         
+  3) external device child1: Uploads configuration file to the thin-edge device with the HTTP filetransfer feature.
+  
+    HTTP PUT request: `http://<ip address of thin-edge devicee>/tedge/tmpfiles`
+    
+  4) The response from the HTTP filetransfer feature contains a _temporary URL_, that is later used to uniquely access that uploaded file.<br/>
+     Example for a _temporary URL_: `/tedge/tmpfiles/<random string>`
+
+  5) external device child1: notifies the the plugin via MQTT about succeeded upload and the _temporary URL_.
+ 
+    Topic: `tedge/configuration/res/retrieve/{config type}/{childid}`<br/>
+    Example: `tedge/configuration/res/retrieve/bar.conf/child1`<br/>
+    Payload: ` <temporary URL> `
+       
+    TODO: Investigate and decide about topic structure and payload in scheduled prototype (https://github.com/thin-edge/thin-edge.io/issues/1307). Unhappy paths also to be considered here.
+  
+  6) C8Y config plugin: recognizes the MQTT notification about the uploaded file and the _temprary URL_, 
+                       and downloads the file using HTTP the filetransfer with the _temporary URL_ to some temporary location.
+  
+    Example for HTTP GET request: `http://<ip address of thin-edge device>/<temporary URL>`
+
+  7) C8Y config plugin: sends the downloaded file to C8Y
+
+  8) C8Y config plugin: removes the downloaded file from temporary location
+  
+
+## Pushing configuration file update to child device from cloud
+
+A configuration file snapshot is pushed to an external device as illustrated below.
+
+```mermaid
+sequenceDiagram
+    participant External Device
+    participant Tedge Agent
+    participant C8Y Cfg Plugin
+    participant C8Y Cloud
+    participant WebSpace
+    
+
+        C8Y Cloud->>C8Y Cfg Plugin: 1: ConfigDownloadRequest (type="bar1", childid="child1", url="some.url")
+
+        C8Y Cfg Plugin->>WebSpace: 2: Download new config-file (url)
+        C8Y Cfg Plugin->>Tedge Agent: 3: HTTP PUT /tedge/tmpfiles (content of new config-file)
+        Tedge Agent->>C8Y Cfg Plugin: 4: HTTP Response, contains tmp-URL, used to access uploaded file
```

Comment:
It is response, so dashed line fits better.
```suggestion
        Tedge Agent-->>C8Y Cfg Plugin: 4: HTTP Response, contains tmp-URL, used to access uploaded file
```

Reply:
Applied.